internal/metrics/metrics.go

```diff
@@ -223,6 +238,46 @@ func (m *Metrics) SetDAGLastRebuilt(ts time.Time) {
 	m.dagRebuildGauge.WithLabelValues().Set(float64(ts.Unix()))
 }
 
+// RecordOnAdd records a successful onAdd event for obj's kind.
+func (m *Metrics) RecordOnAdd(obj interface{}) {
+	m.recordOp("onAdd", obj)
+}
+
+// RecordOnUpdate records a successful onUpdate event for obj's kind.
+func (m *Metrics) RecordOnUpdate(obj interface{}) {
+	m.recordOp("onUpdate", obj)
+}
+
+// RecordOnDelete records a successful onDelete event for obj's kind.
+func (m *Metrics) RecordOnDelete(obj interface{}) {
+	m.recordOp("onDelete", obj)
+}
+
+func (m *Metrics) recordOp(op string, obj interface{}) {
+	m.eventHandlerOperationGauge.WithLabelValues(op, getKind(obj)).Inc()
+}
+
+// getKind returns the obj's kind, or "unknown" if the object
+// does not have a kind.
+func getKind(obj interface{}) string {
+	type typeMetaAccessor interface {
+		GroupVersionKind() schema.GroupVersionKind
+	}
+	switch obj := obj.(type) {
+	case typeMetaAccessor:
+		gvk := obj.GroupVersionKind()
+		kind := fmt.Sprintf("%s/%s%s", gvk.Group, gvk.Kind, gvk.Version)
```

Comment:
This will print `projectcontour/HTTPProxyv1`, right? Or does `Version` have some separator?. Suggest we format "$group/$version:$kind"?

Reply:
yeah, that looks like crap. What about group/version.kind ?

Reply:
Yeh that seems OK too. Not sure if there is a common convention here, but we will have to live with the choice until we retire this metric. Note that for core types `gvk.Group` will be empty.

Reply:
I really f'ing hate the k8s client. Its inconsistencies are legion